packages/views/search.py

```diff
@@ -109,8 +109,9 @@ class SearchListView(ListView):
     paginate_by = 100
 
     sort_fields = ("arch", "repo", "pkgname", "pkgbase", "compressed_size",
-            "installed_size", "build_date", "last_update", "flag_date")
-    allowed_sort = list(sort_fields) + ["-" + s for s in sort_fields]
+                   "installed_size", "build_date", "last_update", "flag_date")
+    allowed_sort = set(sort_fields)
+    allowed_sort.update("-" + s for s in sort_fields)
```

Comment:
``allowed_sort`` is only used to check if something is in it in line 130 ``if sort in self.allowed_sort:`` so it was O(N) before now is O(1) since ``set`` is used albeit the size due to hashing means that more memory is used.

Reply:
This doesn't seem to be the case with Python3, it even looks like lists are faster then
```
>>> timeit.timeit(stmt='10**6 in a', setup='a = range(10**6)', number=100000)
0.011222774000088975
>>> timeit.timeit(stmt='10**6 in a', setup='a = set(range(10**6))', number=100000)
0.002415693999864743
```

Reply:
I think thats not the right benchmark try ``timeit.timeit(stmt='10**6 in a', setup='a = list(range(10**6))', number=100000)``

Reply:
Ah yeah, hmm range() does do some magic then :)

Reply:
Its a generator. I always marvel at the low cost of combining them like ``a = (x for x in something)`` and then use ``a`` somewhere, from some testing I did it is basically free to compose function like this (not completely but it can increase readibility a lot.

But I just realized that range is also slower then set. IIRC set entirely implemented in C.

Reply:
And also to be fair we tested the worst case with a lot of elements this test for example shows that list is better ``timeit.timeit(stmt='0 in a', setup='a = list(range(10**6))', number=100000)`` and in this case list is small so it should do better.

```
In [18]: timeit.timeit(stmt='"arch" in a', setup='a = set(("arch", "repo", "pkgname", "pkgbase", "co
    ...: mpressed_size", "installed_size", "build_date", "last_update", "flag_date"))', number=10000
    ...: 0)
Out[18]: 0.0014225160011847038

In [19]: timeit.timeit(stmt='"arch" in a', setup='a = ["arch", "repo", "pkgname", "pkgbase", "compre
    ...: ssed_size", "installed_size", "build_date", "last_update", "flag_date"]', number=100000)
Out[19]: 0.0013926770006946754
```

I guess this was a useless change...